hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BackgroundService.java

```diff
@@ -62,11 +56,11 @@ public BackgroundService(String serviceName, long interval,
     this.interval = interval;
     this.unit = unit;
     this.serviceName = serviceName;
-    this.serviceTimeout = serviceTimeout;
+    this.serviceTimeoutInNanos = TimeDuration.valueOf(serviceTimeout, unit)
+            .toLong(TimeUnit.NANOSECONDS);
```

Comment:
Usually long is not enough to contain NANOSECONDS but this variable seems just a service timeout, which should be a very small number so it should be fine.